### DISCUSSION TITLE
Setting Seed for HeteroConv Initialization Weights

### DISCUSSION THREAD

**Discussion by yogusmalikombi** [pyg-team/pytorch_geometric#9475] [Q&A]
Hi all,

I'm trying to produce the same initialization weights on  HeteroConv layers. Whenever I run the below code, my weights are initialized differently. I work on CPU, any suggestions?

```
import torch
from torch_geometric.seed import seed_everything
from torch_geometric.nn import SAGEConv,HeteroConv

class HeteroGraphSAGE(torch.nn.Module):
    def __init__(self, hidden_channels, out_channels):
        super(HeteroGraphSAGE, self).__init__()
        self.conv1 = HeteroConv({
            ('user', 'interacts', 'item'): SAGEConv((-1, -1), hidden_channels),
            ('item', 'interacts', 'user'): SAGEConv((-1, -1), hidden_channels),
            ('user', 'interacts', 'user'): SAGEConv((-1, -1), hidden_channels),  
        }, aggr='mean')
        self.conv2 = HeteroConv({
            ('user', 'interacts', 'item'): SAGEConv((hidden_channels, hidden_channels), out_channels),
            ('item', 'interacts', 'user'): SAGEConv((hidden_channels, hidden_channels), out_channels),
            ('user', 'interacts', 'user'): SAGEConv((hidden_channels, hidden_channels), out_channels), 
        }, aggr='mean')
    def forward(self, x_dict, edge_index_dict):
        x_dict = self.conv1(x_dict, edge_index_dict)
        x_dict = {key: F.relu(x) for key, x in x_dict.items()}
        x_dict = self.conv2(x_dict, edge_index_dict)
        return x_dict
    
seed_everything(2666)
model = HeteroGraphSAGE(16,16)
for params in model.parameters():
    print(params)
```
    
Thanks for your replies.

**Comment by EdisonLeeeee** [comment]
Actually, your parameters are not initialized yet. The parameters are [UninitializedParameter](https://pytorch.org/docs/stable/generated/torch.nn.parameter.UninitializedParameter.html) with empty values before running `model.forward`.
